src/lib/construct/staticSite.ts

```diff
@@ -130,6 +132,21 @@ export class StaticSite extends CommonConstruct {
     this.siteOrigin = new origins.S3Origin(this.siteBucket)
   }
 
+  /**
+   * @summary Method to create a site cloudfront function
+   * @protected
+   */
+  protected createSiteCloudfrontFunction() {
+    if (this.props.siteCloudfrontFunctionProps && this.props.siteFunctionFilePath) {
+      this.siteCloudfrontFunction = this.cloudFrontManager.createCloudfrontFunction(
+        `${this.id}-function`,
+        this,
+        this.props.siteCloudfrontFunctionProps,
+        this.props.siteFunctionFilePath
```

Comment:
we shouldn't be getting just the filePath but the whole props from the underlying construct to keep things generic and offer more flexibility